pxtblocks/fields/field_gridpicker.ts

```diff
@@ -65,71 +69,45 @@ namespace pxtblockly {
         }
 
         /**
-         * Create a dropdown menu under the text.
-         * @private
+         * Create blocklyGridPickerRows and add them to table container
+         * @param options 
+         * @param tableContainer 
          */
-        public showEditor_() {
-            Blockly.WidgetDiv.show(this, this.sourceBlock_.RTL, null);
-
+        private populateTableContainer(options: (Object | String[])[], tableContainer: goog.ui.Control) {
             this.disposeTooltips();
-
-            const options = this.getOptions();
-
-            // Container for the menu rows
-            const tableContainer = new goog.ui.Control();
-
-            // Container used to limit the height of the tableContainer, because the tableContainer uses
-            // display: table, which ignores height and maxHeight
-            const scrollContainer = new goog.ui.Control();
-
-            // Needed to correctly style borders and padding around the scrollContainer, because the padding around the
-            // scrollContainer is part of the scrollable area and will not be correctly shown at the top and bottom
-            // when scrolling
-            const paddingContainer = new goog.ui.Control();
-
+            tableContainer.removeChildren(true);
```

Comment:
One thing we should do is test these changes on a very low end laptop, to see if discarding the whole thing and recreating the rows are a performance issue